front/app/containers/Admin/reporting/containers/ReportLanguageProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+
+// Utils
+import { IntlProvider } from 'react-intl';
+import { isNilOrError, NilOrError } from 'utils/helperUtils';
+import moment from 'moment';
+
+// Typings
+import { Locale } from 'typings';
+
+type props = {
+  reportLocale: Locale | NilOrError;
+  platformLocale: Locale | NilOrError;
+  children: React.ReactNode;
+};
+
+// Loads correct messages for the report locale & change moment locale
+const ReportLanguageProvider = ({
+  reportLocale,
+  platformLocale,
+  children,
+}: props) => {
+  const [messages, setMessages] = useState();
+
+  useEffect(() => {
+    if (!isNilOrError(reportLocale)) {
+      // Note: seems inefficient when we have already loaded all these in the main LanguageProvider
+      import(`i18n/${reportLocale}`).then((translationMessages) => {
```

Comment:
Did you check if these imports are cached? In any case, might not be a huge problem since it will only have to be fetched once (when you load the report)

Reply:
Yes these appear to be cached from when they are loaded in the main LanguageProvider